src/main/java/org/codebusters/audiogeek/spotifygazer/application/rest/getreleases/GetReleasesRestAdapter.java

```diff
@@ -0,0 +1,139 @@
+package org.codebusters.audiogeek.spotifygazer.application.rest.getreleases;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.codebusters.audiogeek.spotifygazer.application.rest.getreleases.exception.GetReleasesApiException;
+import org.codebusters.audiogeek.spotifygazer.domain.dataexchange.model.GetAlbumsByGenreFilter;
+import org.codebusters.audiogeek.spotifygazer.domain.newreleases.model.Album;
+import org.codebusters.audiogeek.spotifygazer.domain.newreleases.model.Artist;
+import org.codebusters.audiogeek.spotifygazer.infrastructure.dataexchange.DatabaseDataQueryAdapter;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toSet;
+import static org.codebusters.audiogeek.spotifygazer.application.rest.getreleases.exception.GetReleasesApiErrorData.*;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.ResponseEntity.ok;
+
+@RestController
+@RequestMapping("/api/v1")
+@Slf4j
+@RequiredArgsConstructor
+class GetReleasesRestAdapter {
+    private final GetReleasesProperties props;
+    private final DatabaseDataQueryAdapter query;
+
+    @PostMapping(value = "/get-current-releases", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+    public ResponseEntity<GetReleasesResponse> getCurrentReleases(@RequestBody GetReleasesRequest request,
+                                                                  @RequestParam(name = "page", defaultValue = "0") int page,
+                                                                  @RequestParam(name = "size", defaultValue = "${gazer.app.max-page-size}") int size) {
+        log.info("Processing POST request /get-current-releases: request={}, page={}, size={}", request, page, size);
+        checkPageable(page, size);
+        checkGenres(request.genres());
+        var albumsPage = query.getAlbums(new GetAlbumsByGenreFilter(request.genres(), PageRequest.of(page, size)));
+        var total = albumsPage.getTotalElements();
+        var albums = albumsPage.stream().map(this::convertToOutsideSchema).collect(toSet());
+        var response = GetReleasesResponse.builder()
+                .albums(albums)
+                .page(page)
+                .size(size)
+                .total(total)
+                .build();
+        log.debug("Successfully processed request: response={}", response);
```

Comment:
We have info on the beginning of processing. But debug on end. Please reconsider it. Maybe at least info with less details than debug?